source/includes/_feature-arrays.md

```diff
@@ -14,47 +14,77 @@ Pepsi       o           o      o      o       o
+            {"id":  2, "name": "Not much",   "numeric_value": null, "missing": false},
+            {"id":  3, "name": "OK",         "numeric_value": null, "missing": false},
+            {"id":  4, "name": "A bit",      "numeric_value": null, "missing": false},
+            {"id":  5, "name": "A lot",      "numeric_value": null, "missing": false},
+            {"id": 99, "name": "Skipped",    "numeric_value": null, "missing":  true}
+        ],
+        "values": [
+            [1, 2, {"?": 99}],
+            [{"?": -1}, 4, 3],
+            [5, 2, {"?": -1}],
+        ]
+    }
+}
 ```
 
-These variables would be populated like any other Variable of type "categorical". For example, when sending data Column-by-column, you might POST a Variable Entity with the payload:
+The "Soft Drinks" categorical array variable may now be included in analyses like any other variable, but has 2 dimensions instead of the typical 1. For example, a crosstab of a 1-dimensional "Gender" variable with a 1-dimensional "Education" variable yields a 2-D cube. A crosstab of 1-D "Gender" by 2-D "Soft Drinks" yields a 3-D cube.
+
+In rare cases, you may have already added a separate Variable for "Coke", one for "Pepsi", and one for "RC". You may move them to a single array variable by POSTing a Variable Entity for the array that instead of a "subreferences" attribute has a "subvariables" attribute, a list of URL's of the variables you'd like to bind together:
 
 ```json
 {
-    ...
     "body": {
-        "name": "Coke",
-        "type": "categorical",
-        "categories": [...],
-        "values": [2, 4, 4, {"?": 99}, {"?": -1}, 3, ...]
+        "name": "Soft Drinks",
+        "type": "categorical_array",
+        "subvariables": [<URI of the "Coke" variable>, <URI of the "Pepsi" variable>, <URI of the "RC" variable>]
     }
 }
 ```
 
-Once all of the individual variables have been defined (you don't have to have all their data added yet), then you may create the parent array variable. Doing so by POSTing a Variable Entity would have a payload like:
+The existing variables are removed from the normal order and become virtual subvariables of the new array. This approach will cast all subvariables to a common set of categories if they differ. The existing name and alias of each subvariable will be moved to the array's "subreferences" attribute.
+
+If you wish to analyze a set of categorical variables as an array *without* moving them, you need to build a derived array instead.
 
 ```json
 {
     "body": {
         "name": "Soft Drinks",
         "type": "categorical_array",
-        "subvariables": [<URI of the "Coke" variable>, <URI of the "Pepsi" variable>, <URI of the "RC" variable>]
+        "derivation": {
+            "function": "array",
+            "args": [{
+                "function": "select",
+                "args": [{"map": {
+                    "000000": {"variable": <URI of the "Coke" variable>},
```

Comment:
In "map", is there a necessary relationship between the ids on the outside and the variables they map to? i.e. is "000000" the id of the "Coke" variable?

If not, and it the user is thus manufacturing ids here, do we ensure that they're unique?